pkg/proxy/iptables/proxier.go

```diff
@@ -1292,13 +1394,24 @@ func (proxier *Proxier) syncProxyRules() {
 	natLines := append(natChains.Bytes(), natRules.Bytes()...)
 	lines := append(filterLines, natLines...)
 
-	glog.V(3).Infof("Restoring iptables rules: %s", lines)
-	err = proxier.iptables.RestoreAll(lines, utiliptables.NoFlushTables, utiliptables.RestoreCounters)
+	filterSame, err := sameIptablesRuleset(utiliptables.TableFilter, filterLines, iptablesFilterSaveRaw)
 	if err != nil {
-		glog.Errorf("Failed to execute iptables-restore: %v", err)
-		// Revert new local ports.
-		revertPorts(replacementPortsMap, proxier.portsMap)
-		return
+		glog.Warningf("failed to parse iptables filter rules: %v", err)
```

Comment:
@danwinship the proxy code is the only one so far that uses ParseTableRules(), and while the others could use this too, they don't get called anywhere near as often so they aren't a hotpath like the proxy stuff is.  I'll definitely add comments.